src/spiral_stc.cpp

```diff
@@ -237,19 +232,29 @@ namespace full_coverage_path_planner
         multiple_pass_counter--; // First point is already counted as visited
       }
 
-#ifdef DEBUG_PLOT
-      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Grid with path marked as visited is:");
-      gridNode_t SpiralStart = pathNodes.back();
-      printGrid(grid, visited, pathNodes, pathNodes.front(), pathNodes.back());
-#endif
+      gridNode_t lastNodeAstar = pathNodes.back();
 
       // Spiral fill from current position
       pathNodes = spiral(grid, pathNodes, visited);
 
-#ifdef DEBUG_PLOT
-      RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Visited grid updated after spiral:");
-      printGrid(grid, visited, pathNodes, SpiralStart, pathNodes.back());
-#endif
+      // Need to extract only the spiral part for visualization
+      std::list<gridNode_t> pathNodes_copy = pathNodes; // Work with a copy so that splice() does not affect pathNodes
+      std::list<gridNode_t>:: iterator it = pathNodes_copy.begin();
+      for (const auto node : pathNodes_copy)
+      {
+        if (node.pos.x == lastNodeAstar.pos.x && node.pos.y == lastNodeAstar.pos.y)
+        {
+          break; // Break if iterator is found that matches last node of A* path
+        }
+        ++it; // Iterate one further to find starting point of new spiral
+      }
+      std::list<gridNode_t> pathNodes_spiral;
+      pathNodes_spiral.splice(pathNodes_spiral.begin(), pathNodes_copy, it, pathNodes_copy.end());
```

Comment:
If no iterator is found, this `it` will be larger than the size of `pathNodes_copy`. What happens in that case?

Reply:
I see, it can grow larger than the actual size of pathNodes_copy which might lead to a segfault when used. I now implemented the solution which breaks the for loop when the end iterator of pathNodes_copy is encountered, skipping the ++it line.